### PR TITLE
[WIP] Merge connection packages

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR merges connection packages from autonomy repo and aea repo

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -74,8 +74,7 @@ commands =
     python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-cosmos
     python -m pip install --no-deps file://{toxinidir}/plugins/aea-ledger-fetchai
     python -m pip install --no-deps file://{toxinidir}/plugins/aea-cli-ipfs
-    ; aea test packages {posargs}
-    /bin/bash
+    aea test packages {posargs}
```
Comment (angrybayblade) [inline tox.ini:78]: This has been fixed on the base branch as well

File: packages/valory/connections/http_client/connection.py [hunk 2]
```diff
@@ -373,11 +376,7 @@ class HTTPClientConnection(Connection):
     connection_id = PUBLIC_ID
 
     def __init__(self, **kwargs: Any) -> None:
-        """
```
Comment (DavidMinarsch) [inline packages/valory/connections/http_client/connection.py:376]: Hm the left is more accurate :)

File: packages/valory/connections/http_client/connection.py [hunk 3]
```diff
@@ -292,7 +295,7 @@ def _task_done_callback(self, task: Task) -> None:
         self._tasks.remove(task)
         self.logger.debug(f"Task completed: {task}")
 
-    async def get_message(self) -> Optional["Envelope"]:
+    async def get_message(self) -> Union["Envelope", None]:
```
Comment (DavidMinarsch) [inline packages/valory/connections/http_client/connection.py:298]: Left more accurate :)

**Review by Adamantios — APPROVED** [review]

**Comment by Karrenbelt** [comment]
`FAILED tests/test_mail/test_base.py::test_outbox_put_message`

This test seems nonsensical to me, but what's even more surprising is that it failed.

https://github.com/valory-xyz/open-aea/blob/95e0b7be7081bb92e0fac3cb9cfff6b55755e5c3/tests/test_mail/test_base.py#L194-L219

File: packages/fetchai/skills/erc1155_deploy/tests/test_handlers.py [hunk 4]
```diff
@@ -307,6 +309,7 @@ def test_teardown(self):
         self.assert_quantity_in_outbox(0)
 
 
+@pytest.mark.skip  # Fails because of change in the ledger api protocol
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/erc1155_deploy/tests/test_handlers.py:312]: Please open an issue

**Review by DavidMinarsch — APPROVED** [review]

**Review by Karrenbelt — APPROVED** [review]
There are various failures on windows, not sure why but not observed in child PR #346 

I opened an issue based on another failure: #349

**Pull request merged**